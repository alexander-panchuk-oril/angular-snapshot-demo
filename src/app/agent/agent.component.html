<div class="card">
    <h4>{{ agent.codeName }}</h4>
    <button class="primary">Hire</button>
</div>