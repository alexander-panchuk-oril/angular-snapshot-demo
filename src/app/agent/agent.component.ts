import { Component, Input, OnInit } from '@angular/core';
import { ISecretAgent } from '../shared/interfaces/ISecretAgent.interface';

@Component({
  selector: 'app-agent',
  templateUrl: './agent.component.html',
  styleUrls: ['./agent.component.scss']
})
export class AgentComponent implements OnInit {

  @Input()
  public layout: 'ltr' | 'rtl';

  @Input()
  public agent: ISecretAgent;

  constructor() { }

  ngOnInit(): void {
  }

}
