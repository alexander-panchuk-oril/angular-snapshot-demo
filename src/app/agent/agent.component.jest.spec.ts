import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { AgentComponent } from './agent.component';

describe('AgentComponent', () => {
  let component: AgentComponent;
  let fixture: ComponentFixture<AgentComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ AgentComponent ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(AgentComponent);
    component = fixture.componentInstance;
  });

  it('should renader h4 ', () => {
    const agent =  {
      codeName: 'Velvet Thunder',
      agency: 'Alpha Protocol'
    };
    component.agent = agent;
    fixture.detectChanges();

    const h4 = fixture.debugElement.nativeElement.querySelector('h4');
    expect(h4.innerHTML).toEqual('Velvet Thunder');
  });

  it('should render button ', () => {
    const agent =  {
      codeName: 'Velvet Thunder',
      agency: 'Alpha Protocol'
    };
    component.agent = agent;
    fixture.detectChanges();

    const button = fixture.debugElement.nativeElement.querySelector('button');
    expect(button.innerHTML).toEqual('Hire');
  });

  it('should render h4 before button', () => {
    const agent =  {
      codeName: 'Velvet Thunder',
      agency: 'Alpha Protocol'
    };
    component.agent = agent;
    fixture.detectChanges();

    const h4 = fixture.debugElement.nativeElement.querySelector('h4');
    const button = fixture.debugElement.nativeElement.querySelector('button');
    expect(h4.nextElementSibling).toEqual(button);
  });

  it('should render button after h4', () => {
    const agent =  {
      codeName: 'Velvet Thunder',
      agency: 'Alpha Protocol'
    };
    component.agent = agent;
    fixture.detectChanges();

    const h4 = fixture.debugElement.nativeElement.querySelector('h4');
    const button = fixture.debugElement.nativeElement.querySelector('button');
    expect(button.previousElementSibling).toEqual(h4);
  });

  it('[snapshot] should renders markup to snapshot', () => {
    const agent =  {
      codeName: 'Velvet Thunder',
      agency: 'Alpha Protocol'
    };
    component.agent = agent;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
