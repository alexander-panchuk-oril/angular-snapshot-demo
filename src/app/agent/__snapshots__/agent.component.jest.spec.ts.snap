// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AgentComponent [snapshot] should renders markup to snapshot 1`] = `
<app-agent
  agent={[Function Object]}
>
  <div
    class="card"
  >
    <h4>
      Velvet Thunder
    </h4>
    <button
      class="primary"
    >
      Hire
    </button>
  </div>
</app-agent>
`;
